apps/extension/src/app/services/use-account-addresses.ts

```diff
@@ -1,19 +1,25 @@
 import { useMemo } from 'react';
 
+import { addDerivedAddressesToAccountAddresses } from '@leather.io/bitcoin';
 import { createAccountAddresses } from '@leather.io/utils';
 
 import { useBitcoinAccountXpubs } from '@app/store/accounts/blockchain/bitcoin/bitcoin.hooks';
 import { useStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { useCurrentNetwork } from '@app/store/networks/networks.selectors';
 
 export function useAccountAddresses(accountIndex: number) {
   const accountXpubs = useBitcoinAccountXpubs(accountIndex);
   const stxAccount = useStacksAccount(accountIndex);
+  const currentNetwork = useCurrentNetwork();
 
   return useMemo(() => {
-    return createAccountAddresses(
-      { fingerprint: 'master', accountIndex },
-      accountXpubs,
-      stxAccount?.address
+    return addDerivedAddressesToAccountAddresses(
```

Comment:
I'm also adding the derived addresses to accounts on the extension as I plan to use this service there also soon but I'm unsure if that's necessary there.

Reply:
I have difficult understanding what `addDerivedAddressesToAccountAddresses` means

Reply:
I've changed this to use the existing hooks to add to the `accountAddressesSchema`. I initially misunderstood and thought I would need to pass them to the collectibles service directly. 

I've changed it to keep a similar approach but to use existing hooks instead

Reply:
Thanks both for the feedback, it has helped me learn a lot. I am going to merge this but we can consider a refactor of `AccountAddresses` soon after? I can do it right away if deemed urgent.